haystack/preview/components/file_converters/txt.py

```diff
@@ -0,0 +1,229 @@
+                                This option can be used to add a test for encoding errors. If the extracted text is
+                                not one of the valid languages, then there might be an encoding error resulting
+                                in garbled text.
+                                Default: `None`
+        :param id_hash_keys: Generate the Document ID from a custom list of strings that refer to the Document's
+                             attributes. If you want to ensure you don't have duplicate Documents in your DocumentStore
+                             but texts are not unique, you can modify the metadata and pass e.g. `"meta"` to this field
+                             (for example `["content", "meta"]`). In this case the ID will be generated by using the
+                             content and the defined metadata. Default: `None`
+        :param progress_bar: Whether to show a progress bar for the conversion process.
+                             Default: `True`
+        """
+        langdetect_import.check()
+
+        self.defaults = {
+            "encoding": encoding,
+            "remove_numeric_tables": remove_numeric_tables,
+            "valid_languages": valid_languages,
+            "id_hash_keys": id_hash_keys,
+            "progress_bar": progress_bar,
+        }
+
+    def run(self, data):
+        """
+        Convert text files to Documents.
+
+        :param data: Input data for the TextfileToDocument component.
+        """
+        file_paths = data.paths
+        metas = TextfileToDocument._prepare_metadata(data.meta, file_paths)
+
+        documents = []
+        for path, meta in zip(file_paths, metas):
+            text = TextfileToDocument._read_and_clean_file(
+                path=path, encoding=data.encoding, remove_numeric_tables=data.remove_numeric_tables
+            )
+
+            if data.valid_languages is not None:
+                if not TextfileToDocument._validate_language(text, data.valid_languages):
+                    logger.warning(
+                        f"Text from file {path} is not in one of the valid languages: {data.valid_languages}. "
+                        f"The file may have been decoded incorrectly."
+                    )
+
+            document = Document(content=text, meta=meta, id_hash_keys=data.id_hash_keys)
+            documents.append(document)
+
+        return self.output(documents=documents)
+
+    @staticmethod
+    def _prepare_metadata(meta: Optional[Union[Dict, List[Dict]]], file_paths: List[Union[str, Path]]) -> List[Dict]:
+        """
+        Prepare the metadata for the Documents.
+
+        :param meta: The metadata for the Documents.
+        :param file_paths: The paths to the text files.
+        """
+        if meta is None:
+            return [{"file_path": str(path)} for path in file_paths]
```

Comment:
Here I was not sure if we want to store the whole provided file path in the document's meta data or just the file name.

Reply:
Since you can derive file name from the full path, perhaps that's a better option?